src/useQuery.ts

```diff
@@ -161,10 +175,17 @@ export function useQuery<TData = any, TVariables = OperationVariables>(
     };
   }
 
-  if (suspend && currentResult.partial) {
-    // throw a promise - use the react suspense to wait until the data is
-    // available
-    throw observableQuery.result();
+  if (currentResult.partial) {
+    if (suspend) {
```

Comment:
I'd change the order here - SSR should work also for queries using the default suspense mode (at least until suspense-ready official React server renderer is available). So rather:

```typescript
  if (currentResult.partial) {
    // Register request only when `ssr: true`.
    if (ssrInUse) {
      ssrManager.register(observableQuery.result());
    }

    else if (suspend) {
      // throw a promise - use the react suspense to wait until the data is
      // available
      throw observableQuery.result();
    }
  }
```

Also tests should check both suspense and non-suspense mode.

Reply:
Suspense does not work in react-dom

Reply:
> Suspense does not work in react-dom

I'm not sure I understand. I changed the code to the one in the snipped above and removed `suspend: false` from `useQuery` invocations in _getMarkupFromTree-test.tsx_ and tests passed. Or do you mean that the `<Suspense />` component is not supported by the current version of react-dom-server renderer? We could workaround it by creating our custom helper component and recommending to use it instead of `<Suspense />` from React:

```typescript
export const SuspenseSsr = (props: SuspenseProps) => {
  const ssrManager = useContext(SSRContext);
  if (ssrManager) {
    return <>{props.children}</>;
  }

  return (<Suspense {...props} />)
}
```

Reply:
i think it’s better to force developers pass suspend: false if they want to use ssr, otherwise they will have different behavior on ssr and client side.
suspense assures user that data is not null on render, and if ignore suspense on ssr - this can create randomnull pointer errors. Current approach will escalate suspended promises to react-dom and it will give good error measage.

Reply:
oops, i’m writing from phone and didn’t saw your last message.

but my previous message also covers your comment.

from my experience when you’re writing ssr app - you have to write it ssr first otherwise it would be hacky and buggy. it’s better to wait for officials suspense support from react. if someone REALLY wants to use suspense for components that also rendered on server and deal with all the problem it produeses - they can write their own suspense component

Reply:
girhub mobile is killing me 🤦‍♂️, it adds teo comments at once and freezes when i want to remove one

Reply:
> wouldn’t it better to name it “unstable_SuspenseSSR” then 🤔.

👍 

> i like the idea, i’ll try to implement it tonight

Thank you :)

Reply:
@trojanowski I could't wait 🙂. So I start catching thrown promises in `getMarkupFromTree` and add tests for `suspend: true`.

Reply:
I can not wait too 👍 thanks guys

Reply:
@umidbekkarimov Looks great, thanks :) I added a few comments about testing.